### PR TITLE
curvefs: Merge updateInode and GetOrModifyS3ChunkInfo 2 rpcs in backg…

### PR DESCRIPTION
**Pull request by baijiaruo**
…round flush process

Signed-off-by: baijiaruo <baijiaruo@126.com>

### What problem does this PR solve?

Issue Number: #1361  

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/client/inode_cache_manager.cpp [hunk 1]
```diff
@@ -348,33 +407,53 @@ void InodeCacheManagerImpl::ReleaseCache(uint64_t parentId) {
...
         }
+        FlushInodeOnce();
+        sleeper_.wait_for(std::chrono::seconds(flushPeriodSec_));
     }
     LOG(INFO) << "flush thread is stop.";
 }
 
+void InodeCacheManagerImpl::RemoveICache(
+    const std::shared_ptr<InodeWrapper> &inode) {
+    if (!inode->IsDirty() && inode->S3ChunkInfoEmpty()) {
```
Comment (xu-chaojie) [inline curvefs/src/client/inode_cache_manager.cpp:421]: Here should get lock before judging IsDirty and S3ChunkInfoEmpty

**Review by xu-chaojie — CHANGES_REQUESTED** [review]

File: curvefs/src/client/inode_cache_manager.cpp [hunk 2]
```diff
@@ -348,33 +407,53 @@ void InodeCacheManagerImpl::ReleaseCache(uint64_t parentId) {
...
+    VLOG(9) << "TrimIcache trimSize " << trimSize;
     while (trimSize > 0) {
-        bool ok = iCache_->GetLast(&inodeId, &inodeWrapper, IsNotDirtyInode);
+        bool ok = iCache_->GetLast(&inodeId, &inodeWrapper);
         if (ok) {
-            NameLockGuard lock(nameLock_, std::to_string(inodeId));
-            ::curve::common::UniqueLock lgGuard = inodeWrapper->GetUniqueLock();
-            if (!inodeWrapper->IsDirty() &&
-                inodeWrapper->S3ChunkInfoEmptyNolock()) {
+            if (inodeWrapper->IsDirty() || !inodeWrapper->S3ChunkInfoEmpty()) {
```
Comment (xu-chaojie) [inline curvefs/src/client/inode_cache_manager.cpp:437]: Here should lock before judging IsDirty and S3ChunkInfoEmpty

File: curvefs/src/client/inode_wrapper.cpp [hunk 3]
```diff
@@ -397,21 +396,77 @@ CURVEFS_ERROR InodeWrapper::UpdateParentLocked(
...
+        case FsFileType::TYPE_FILE:
+            ret = SyncAttr(internal);
+            if (ret != CURVEFS_ERROR::OK) {
+                break;
+            }
+            ret = FlushVolumeExtent();
+            break;
+        case FsFileType::TYPE_DIRECTORY:
+            ret = SyncAttr(internal);
+            if (ret != CURVEFS_ERROR::OK) {
```
Comment (xu-chaojie) [inline curvefs/src/client/inode_wrapper.cpp:413]: This if  statement is not needed

File: curvefs/src/client/inode_wrapper.cpp [hunk 4]
```diff
@@ -397,21 +396,77 @@ CURVEFS_ERROR InodeWrapper::UpdateParentLocked(
...
+
+        if (ret != MetaStatusCode::OK) {
+            LOG(ERROR) << "metaClient_ UpdateInodeAttrWithOutNlink failed, "
+                       << "MetaStatusCode: " << ret
+                       << ", MetaStatusCode_Name: " << MetaStatusCode_Name(ret)
+                       << ", inodeid: " << inode_.inodeid();
+            dirty_ = true;
+            return MetaStatusCodeToCurvefsErrCode(ret);
+        }
+        s3ChunkInfoAdd_.clear();
```
Comment (xu-chaojie) [inline curvefs/src/client/inode_wrapper.cpp:457]: missing lock of syncing s3ChunkInfo

File: curvefs/src/client/rpcclient/metaserver_client.cpp [hunk 5]
```diff
@@ -959,9 +962,9 @@ void MetaServerClientImpl::UpdateInodeAttrAsync(
 
 void MetaServerClientImpl::UpdateInodeAttrWithOutNlinkAsync(
```
Comment (xu-chaojie) [inline curvefs/src/client/rpcclient/metaserver_client.cpp:963]: UpdateInodeAttrWithOutNlinkAsync  not only update attr also s3info, should rename this function

**Comment by baijiaruo** [comment]
recheck

File: curvefs/src/metaserver/inode_storage.cpp [hunk 6]
```diff
@@ -572,5 +573,22 @@ MetaStatusCode InodeStorage::GetVolumeExtentByOffset(uint32_t fsId,
     return MetaStatusCode::STORAGE_INTERNAL_ERROR;
 }
 
+MetaStatusCode
+InodeStorage::AddS3ChunkInfoList(uint32_t fsId, uint64_t inodeId,
```
Comment (Wine93) [inline curvefs/src/metaserver/inode_storage.cpp:577]: maybe you can use the exist function `AddS3ChunkInfoList` and distinguished by param, if `txn==null`, we use `kvStorage_->SSet()`, else use `txn->SSet()`.

File: curvefs/src/client/inode_cache_manager.cpp (hunk 1 above)
Comment (baijiaruo) [reply curvefs/src/client/inode_cache_manager.cpp:421]: fixed

**Review by xu-chaojie — APPROVED** [review]

**Review by Wine93 — APPROVED** [review]

**Pull request merged**